### PR TITLE
network manager plugins (iodine/vpnc) without gnome

### PR DESCRIPTION
**Pull request by dguibert**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by doronbehar** [comment]
@dguibert this seems interesting but I'm not sure I get the motivation or what this does.

File: pkgs/tools/networking/network-manager/vpnc/default.nix [hunk 1]
```diff
@@ -24,7 +24,8 @@ in stdenv.mkDerivation {
   ];
 
   buildInputs = [ vpnc networkmanager ]
-    ++ stdenv.lib.optionals withGnome [ gtk3 libsecret libnma ];
+    ++ stdenv.lib.optionals withGnome [ gtk3 libsecret libnma ]
+    ++ stdenv.lib.optionals (!withGnome) [ glib ];
```
Comment (jtojnar) [inline pkgs/tools/networking/network-manager/vpnc/default.nix:28]: It should be unconditional:

https://gitlab.gnome.org/GNOME/NetworkManager-vpnc/-/blob/e60f4cf06cd79ee1ec913ec277b3d0c2c90184f7/configure.ac#L98

File: pkgs/tools/networking/network-manager/iodine/default.nix [hunk 2]
```diff
@@ -28,7 +28,8 @@ in stdenv.mkDerivation {
   ];
 
   buildInputs = [ iodine networkmanager ]
-    ++ stdenv.lib.optionals withGnome [ gtk3 libsecret libnma ];
+    ++ stdenv.lib.optionals withGnome [ gtk3 libsecret libnma ]
+    ++ stdenv.lib.optionals (!withGnome) [ glib ];
```
Comment (jtojnar) [inline pkgs/tools/networking/network-manager/iodine/default.nix:32]: Same here: https://gitlab.gnome.org/GNOME/network-manager-iodine/-/blob/2a2aa853aac038aa46411b20ad3bb90f645ad689/src/nm-iodine-service.c#L42

**Comment by dguibert** [comment]
> @dguibert this seems interesting but I'm not sure I get the motivation or what this does.

This patch allows to compile these plugins when setting in your `configuration.nix`
```nix
environment.noXlibs = true
```

**Comment by jtojnar** [comment]
Thanks.

**Pull request merged**